.github/workflows/publish-pypi-deploy.yml

```diff
@@ -12,52 +12,32 @@ jobs:
       fail-fast: true
       matrix:
         python-version: [ '3.10' ]
-
+    environment:
+      name: testpypi  # pypi
+      url: https://test.pypi.org/legacy/  # https://pypi.org/p/pydicom
+    permissions:
+      id-token: write
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout
+      uses: actions/checkout@v3
 
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
 
     - name: Build package
       run: |
         python -m pip install --upgrade pip
-        python -m pip install wheel
-        python setup.py sdist bdist_wheel
-
-    - name: Upload artifacts
-      if: ${{ success() }}
-      uses: actions/upload-artifact@v2
-      with:
-        name: wheels
-        path: dist/*.whl
-
-    - name: Set up venv
-      run: |
-        python -m venv ~/myenv
-        source ~/myenv/bin/activate
-        echo "Virtual env: $VIRTUAL_ENV"
-
-    - name: Install pytest and our package
-      run: |
-        python -m pip install -U pytest
-        cd ..  # avoid pip seeing 'pydicom' package dir below
-        python -m pip install -f pydicom/dist pydicom
-
-    - name: Test with pytest (no numpy)
-      run: |
-        python -c "import pytest; pytest.main(['--pyargs', 'pydicom.tests'])"
-
-    - name: Install numpy and test
-      run: |
-        python -m pip install -U numpy
-        python -c "import pytest; pytest.main(['--pyargs', 'pydicom.tests'])"
-        cd pydicom  # back to root repo dir
-
-    - name: Publish package to PyPi
-      uses: pypa/gh-action-pypi-publish@master
-      with:
-        user: __token__
-        password: ${{ secrets.PYPI_PASSWORD }}
+        python -m pip install flit
+        flit build
+
+    # Run tests?  Removed for now as merge into main does full testing
```

Comment:
Not sure... it would be good to ensure that the workflow only runs if the tests succeeded, and currently there seems to be no easy way to do this if the tests are running in another workflow.

Reply:
> Not sure... it would be good to ensure that the workflow only runs if the tests succeeded, and currently there seems to be no easy way to do this if the tests are running in another workflow.

I think it should be okay. This is triggered by publishing an official release, so at that point we'll know that the last merge into main passed all merge tests.

Reply:
Agreed. We can assume that whoever does the publishing, will ensure that the tests pass before 😄. No need to over-automate things...